src/Concordium/Client/Runner.hs

```diff
@@ -1938,6 +1946,72 @@ processContractCmd action baseCfgDir verbose backend =
                                                 using the function '#{receiveName}'|]]
 
+    ContractInvoke indexOrName subindex receiveName parameterFile schemaFile amount invoker energy block -> do
+      baseCfg <- getBaseConfig baseCfgDir verbose
+      namedContrAddr <- getNamedContractAddress (bcContractNameMap baseCfg) indexOrName subindex
+
+      invoker' :: Maybe Types.Address <- case invoker of
+        Nothing -> return Nothing
+        Just (InvokerAccount nameOrAddr) -> Just . Types.AddressAccount . naAddr <$> getAccountAddressArg (bcAccountNameMap baseCfg) nameOrAddr
+        Just InvokerContract{..} -> Just . Types.AddressContract . ncaAddr <$> getNamedContractAddress (bcContractNameMap baseCfg) icIndexOrName icSubindex
+
+      contrInfo <- withClient backend . withBestBlockHash block $ \bb -> getContractInfo namedContrAddr bb
+      let namedModRef = NamedModuleRef {nmrRef = CI.ciSourceModule contrInfo, nmrNames = []} -- Skip finding nmrNames, as they won't be shown.
+
+      let contractName = CI.getContractName contrInfo
+      let wasmReceiveName = Wasm.ReceiveName [i|#{contractName}.#{receiveName}|]
+      unless (CI.hasReceiveMethod receiveName contrInfo) $ logFatal [[i|The contract does not have a receive function named '#{receiveName}'.|]]
+
+      modSchema <- withClient backend . withBestBlockHash block $ getSchemaFromFileOrModule schemaFile namedModRef
+      wasmParameter <- getWasmParameter parameterFile modSchema (CS.ReceiveFuncName contractName receiveName)
+      let nrg = fromMaybe (Types.Energy 10_000_000) energy
+
+      let invokeContext = InvokeContract.ContractContext
+                          { ccInvoker = invoker'
+                          , ccContract = ncaAddr namedContrAddr
+                          , ccAmount = amount
+                          , ccMethod = wasmReceiveName
+                          , ccParameter = wasmParameter
+                          , ccEnergy = nrg
+                          }
+      invokeContextArg <- case Text.decodeUtf8' . BSL.toStrict . AE.encode $ invokeContext of
+            Left _ -> logFatal ["Could not invoke contract due to internal error: decoding UTF-8 failed"] -- Should never happen.
+            Right text -> return text
+
+      withClient backend . withBestBlockHash block $ \bb -> do
+        res <- invokeContract invokeContextArg bb
+        case res of
+          Left err -> logFatal [[i|Invocation failed with error: #{err}|]]
+          Right jsonValue -> case AE.fromJSON jsonValue of
+            AE.Error jsonErr -> logFatal [[i|Invocation failed with error: #{jsonErr}|]]
+            AE.Success InvokeContract.Failure{..} ->
+              -- Logs in yellow to indicate that the invocation returned with a failure.
+              -- This might be what you expected from the contract, so logWarn or logFatal should not be used.
+              Cli.log Info (Just ANSI.Yellow) [[iii|Invocation resulted in failure:\n
```

Comment:
It might also be worth using a different color than the one for warnings. To make it even clearer.